### PR TITLE
Add notes about creating a custom target 

### PR DESCRIPTION
**Pull request by lights0123**
As we were discussing rust-lang/rust#58500, I realized that there is nowhere other than community blogs that actually discusses *how* to create a new target. This adds new documentation about creating custom targets and common pitfalls that you may encounter.

### CODE REVIEW

**Comment by lights0123** [comment]
This also supersedes #42, as I wrote about eh_personality without looking at existing PRs.

**Comment by lights0123** [comment]
@jamesmunns any chance to get this reviewed?

File: src/custom-target.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+# Creating a custom target
+
+If a custom target triple is not available for your platform, you must create a custom target file
+that describes your target to rustc.
+
+Keep in mind that it is required to use a nightly compiler to build the core library, which must be
+done for a target unknown to rustc.
+
+## Decide on a target triple
```
Comment (therealprof) [inline src/custom-target.md:9]:
```suggestion
## Deciding on a target triple
```

File: src/custom-target.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+Many targets already have a known triple used to describe them, typically in the form
+ARCH-VENDOR-SYS-ABI. You should aim to use the same triple that [LLVM uses][llvm-target-triple];
```
Comment (therealprof) [inline src/custom-target.md:12]: Uhm, ARCH-VENDOR-SYS-ABI is a quadruple(t), not a triple(t). ;)

File: src/custom-target.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+however, it may differ if you need to specify additional information to Rust that LLVM does not know
+about. Although the triple is technically only for human use, it's important for it to be unique and
+descriptive especially if the target will be upstreamed in the future.
+
+The ARCH part is typically just the architecture name, except in the case of 32-bit ARM. For
+example, you would probably use x86_64 for those processors, but specify the exact ARM architecture
+version. Typical values might be `armv7`, `armv5te`, or `thumbv7neon`. Take a look at the names of
+the [built-in targets][built-in-target] for inspiration.
+
+The VENDOR part is optional, and describes the manufacturer. Omitting this field is the same as
```
Comment (therealprof) [inline src/custom-target.md:22]:
```suggestion
The VENDOR part is optional and describes the manufacturer. Omitting this field is the same as
```

**Pull request merged**